src/TimeSteppers/runge_kutta_3.jl

```diff
@@ -132,7 +135,10 @@ function time_step!(model::AbstractModel{<:RungeKutta3TimeStepper}, Δt; callbac
     calculate_pressure_correction!(model, third_stage_Δt)
     pressure_correct_velocities!(model, third_stage_Δt)
 
-    tick!(model.clock, third_stage_Δt)
+    # This formulation of the final time-step reduces the accumulation of round-off error when Δt
+    # is added to model.clock.time.
+    corrected_third_stage_Δt = tⁿ⁺¹ - model.clock.time
+    tick!(model.clock, corrected_third_stage_Δt)
     update_state!(model, callbacks; compute_tendencies)
     step_lagrangian_particles!(model, third_stage_Δt)
```

Comment:
Here we `tick!` with `corrected_third_stage_Δt`, but other third stage steps (namely we `calculate_pressure_correction!`, `pressure_correct_velocities!` and `step_langrangian_particles!`) use the original `third_stage_Δt`. I would assume all of them need to use the `corrected` version, no?

Reply:
hmm we can change that though, though it shouldn't matter because they are indistinguishable except by machine epsilon?

Reply:
Ah right I did consider this. We can change the pressure correction and lagrangian particles. However, other tracers effectively use the original third stage dt via rk3 substep. So I think it's actually simpler to use the same third_state_dt for everything (tracers, predictor velocity, pressure correction, lagrangian particles), but to simply shave off the error accumulated in the model clock during the substeps

Reply:
> hmm we can change that though, though it shouldn't matter because they are indistinguishable except by machine epsilon?

Yeah, it's more a matter of consistency than anything else imo

Reply:
Right, so my point is that it's more consistent to use the non-corrected time-step everywhere, including `rk3substep!` (where it is used implicitly), as well as in the pressure correction and Lagrangian particles step. Do you agree?

Reply:
Ah, I see what you're saying. Yes I agree.